File: Scores/Models/StudentSubject.cs
using System.Collections.Generic;
using System.Text.Json.Serialization;

namespace Scores.Models
{
    public class StudentSubject
    {
        [JsonPropertyName("student_id")]
        public int StudentId { get; set; }

        [JsonPropertyName("name")]
        public string StudentName { get; set; }

        [JsonPropertyName("subject")]
        public string Subject { get; set; }

        [JsonPropertyName("scores")]
        public List<StudentScore> Scores { get; set; }
    }
}

File: scores-php/src/Models/StudentSubject.php
<?php

namespace Scores\Models;

class StudentSubject
{
    public function __construct(
        public int $studentId = 0,
        public string $studentName = '',
        public string $subject = '',
        public array $scores = []
    ) {}
    
    public function toArray(): array
    {
        return [
            'student_id' => $this->studentId,
            'student_name' => $this->studentName,
            'subject' => $this->subject,
            'scores' => array_map(fn($score) => $score->toArray(), $this->scores)
        ];
    }
}

